### PR TITLE
Add exchange to custom address (L1 <=> L2)

### PR DESCRIPTION
**Pull request by kirilantevski**
- Thales to different receiver account on L2 Thales 
- Thales (L1) to different OpThales (L1) receiver account
- OpThales (L1) to different Thales (L1) receiver account

### CODE REVIEW

**Comment by kirilantevski** [comment]
- Solution tested 
- Ready for merging

File: contracts/exchange_L1_L2/ThalesExchanger.sol [hunk 1]
```diff
@@ -123,6 +151,21 @@ contract ThalesExchanger is IThalesExchanger, Initializable, ProxyOwned, ProxyRe
...
+        address _sender,
+        address _receiver,
+        uint _amount,
+        uint _thalesToOpThales
+    ) internal notPaused {
+        if (_thalesToOpThales == THALES_TO_OPTHALES) {
+            ThalesToken.transferFrom(_sender, address(this), _amount);
+            OpThalesToken.transfer(_receiver, _amount);
+        } else {
+            OpThalesToken.transferFrom(_sender, address(this), _amount);
```
Comment (danijelthales) [inline contracts/exchange_L1_L2/ThalesExchanger.sol:165]: we only need this when bridging from l1 to l2, so not needed here

File: contracts/exchange_L1_L2/ThalesExchanger.sol [hunk 2]
```diff
@@ -91,6 +91,15 @@ contract ThalesExchanger is IThalesExchanger, Initializable, ProxyOwned, ProxyRe
         _exchange(msg.sender, amount, THALES_TO_OPTHALES);
         emit ExchangedThalesForOpThales(msg.sender, amount);
     }
+    
+    function exchangeThalesToOpThalesDifferentReceiver(address receiver_account, uint amount) external nonReentrant notPaused {
```
Comment (danijelthales) [inline contracts/exchange_L1_L2/ThalesExchanger.sol:95]: not needed

File: contracts/exchange_L1_L2/ThalesExchanger.sol [hunk 3]
```diff
@@ -99,6 +108,15 @@ contract ThalesExchanger is IThalesExchanger, Initializable, ProxyOwned, ProxyRe
         _exchange(msg.sender, amount, OPTHALES_TO_THALES);
         emit ExchangedOpThalesForThales(msg.sender, amount);
     }
+    
+    function exchangeOpThalesToThalesDifferentReceiver(address receiver_account, uint amount) external nonReentrant notPaused {
```
Comment (danijelthales) [inline contracts/exchange_L1_L2/ThalesExchanger.sol:112]: not needed

File: contracts/exchange_L1_L2/ThalesExchanger.sol [hunk 4]
```diff
@@ -108,6 +126,16 @@ contract ThalesExchanger is IThalesExchanger, Initializable, ProxyOwned, ProxyRe
         L1Bridge.depositERC20To(address(OpThalesToken), l2TokenAddress, msg.sender, amount, 2000000, "0x");
         emit ExchangedThalesForL2OpThales(msg.sender, amount);
     }
+    
+    function exchangeThalesToL2OpThalesDifferentReceiver(address receiver_account, uint amount) external nonReentrant notPaused {
```
Comment (danijelthales) [inline contracts/exchange_L1_L2/ThalesExchanger.sol:130]: this is the only method we need

**Review by danijelthales — CHANGES_REQUESTED** [review]

**Review by danijelthales — APPROVED** [review]

**Pull request merged**